packages/math/src/bn.test.ts

```diff
@@ -280,4 +280,152 @@ describe('Math - Convert', () => {
     expect(bn(maxBytes).toHex(4)).toEqual(maxHex);
     expect(() => bn(over).toHex(4)).toThrow();
   });
+
+  it('should accept possibly undefined to craete bn, and set to 0 when undefined', () => {
+    const inputs: { numb: number; undef?: string } = {
+      numb: 2,
+    };
+
+    expect(bn().toNumber()).toEqual(0);
+    expect(bn(inputs?.undef).toNumber()).toEqual(0);
+    expect(bn(inputs?.numb).toNumber()).toEqual(2);
+  });
+
+  it('should formatUnits from default unit', () => {
+    expect(bn('1000000000').formatUnits()).toEqual('1.000000000');
+    expect(bn('2').formatUnits()).toEqual('0.000000002');
+    expect(bn('20000').formatUnits()).toEqual('0.000020000');
+    expect(bn('100000020000').formatUnits()).toEqual('100.000020000');
+    expect(bn('100100000020000').formatUnits()).toEqual('100100.000020000');
+  });
+
+  it('should formatUnits from supplied unit', () => {
+    expect(bn('1000000000').formatUnits(7)).toEqual('100.0000000');
+    expect(bn('2').formatUnits(7)).toEqual('0.0000002');
+    expect(bn('20000').formatUnits(7)).toEqual('0.0020000');
+    expect(bn('100000020000').formatUnits(7)).toEqual('10000.0020000');
+    expect(bn('100100000020000').formatUnits(7)).toEqual('10010000.0020000');
+  });
+
+  it('should format with default configs', () => {
+    expect(bn('1000000000').format()).toEqual('1');
+    expect(bn('2').format()).toEqual('0.000000002');
+    expect(bn('22000').format()).toEqual('0.00002');
+    expect(bn('100000020000').format()).toEqual('100');
+    expect(bn('100100000020000').format()).toEqual('100,100');
```

Comment:
I think we should keep at least a `.0` not having. `.` and having `,` could lead to the miss understanding of the amount being a decimal.